lib/executor/ramping_arrival_rate.go

```diff
@@ -271,24 +271,34 @@ func (varc RampingArrivalRateConfig) cal(et *lib.ExecutionTuple, ch chan<- time.
 	for _, stage := range varc.Stages {
 		to = float64(stage.Target.ValueOrZero()) / timeUnit
 		dur = float64(stage.Duration.Duration)
-		if from != to { // ramp up/down
-			endCount += dur * ((to-from)/2 + from)
-			for ; i <= endCount; i += float64(next()) {
-				// TODO: try to twist this in a way to be able to get i (the only changing part)
-				// somewhere where it is less in the middle of the equation
-				x := (from*dur - noNegativeSqrt(dur*(from*from*dur+2*(i-doneSoFar)*(to-from)))) / (from - to)
-
-				ch <- time.Duration(x) + stageStart
-			}
-		} else {
+		switch progression := stage.Progression.String; progression {
```

Comment:
This looks like a good refactor opportunity: to have the interpolation functions being dedicated either functions or methods called by the various switch cases. One good hint from my point of view, is that the `constant` branch is basically a duplicate of the non-ramping behavior of the `linear` one.